### PR TITLE
release-notes: draft v10.1.0 release notes

### PR DESCRIPTION
**Pull request by dachary**
Signed-off-by: Loic Dachary <loic@dachary.org>

### CODE REVIEW

**Comment by dachary** [comment]
@liewegas note the last five entries: the body of the merge message may need manual edit.

File: doc/release-notes.rst [hunk 1]
```diff
@@ -2,6 +2,440 @@
...
+    rgw: adjust error code when bucket does not exist in copy operation
+    rgw: don't override error when initializing zonegroup
+    Fixes: #14975
+* Wip rgw indexless (`pr#7786 <http://github.com/ceph/ceph/pull/7786>`_, Yehuda Sadeh)
+    rgw: indexless buckets (Yehuda Sadeh)
+    - can define a policy, for which buckets are indexless
+    - users can then create buckets under the specified placement target
+    - indexless buckets will not be synced across zones
+    - does not work with (s3) versioned buckets
+    Signed-off-by: Yehuda Sadeh <yehuda@redhat.com>
```
Comment (theanalyst) [inline doc/release-notes.rst]: Maybe we drop the signed off by

File: doc/release-notes.rst [hunk 2]
```diff
@@ -2,6 +2,440 @@
...
+* xio: remove redundant magic methods (`pr#7773 <http://github.com/ceph/ceph/pull/7773>`_, Roi Dayan)
+* xio: remove unused variable (`pr#8023 <http://github.com/ceph/ceph/pull/8023>`_, Roi Dayan)
+* xxhash: use clone of xxhash.git; add .gitignore (`pr#7986 <http://github.com/ceph/ceph/pull/7986>`_, Sage Weil)
+* rbd: update default image features (`pr#7846 <http://github.com/ceph/ceph/pull/7846>`_, Jason Dillaman)
+* rbd-mirror: make remote context respect env and argv config params (`pr#8182 <http://github.com/ceph/ceph/pull/8182>`_, Mykola Golub)
+* journal: re-use common threads between journalers (`pr#7906 <http://github.com/ceph/ceph/pull/7906>`_, Jason Dillaman)
+* client: add option to control how directory size is calculated (`pr#7323 <http://github.com/ceph/ceph/pull/7323>`_, Yan, Zheng)
+    This lets you disable rstats if your workload is unhappy about directories
+    changing size (eg, tar of recently-moved/created/untarred files).
+* Wip rgw keystone v3 (`pr#7719 <http://github.com/ceph/ceph/pull/7719>`_, Mark Barnes, Radoslaw Zarzynski)
```
Comment (theanalyst) [inline doc/release-notes.rst]: Drop wip

File: doc/release-notes.rst [hunk 3]
```diff
@@ -2,6 +2,440 @@
...
+    rgw keystone v3 (Mark Barnes, Radoslaw Zarzynski)
+    support for keystone v3
+* Wip rgw new multisite merge (`issue#14549 <http://tracker.ceph.com/issues/14549>`_, `pr#7709 <http://github.com/ceph/ceph/pull/7709>`_, Yehuda Sadeh, Orit Wasserman, Casey Bodley, Daniel Gryniewicz)
```
Comment (theanalyst) [inline doc/release-notes.rst]: Drop wip

File: doc/release-notes.rst [hunk 4]
```diff
@@ -2,6 +2,440 @@
...
+* Add --setuser and --setgroup options for ceph-disk (`pr#7351 <http://github.com/ceph/ceph/pull/7351>`_, Mike Shuey)
+* Adding build requires (`pr#7742 <http://github.com/ceph/ceph/pull/7742>`_, Erwan Velu)
+* AsyncMessenger: fix several bugs (`pr#7831 <http://github.com/ceph/ceph/pull/7831>`_, Haomai Wang)
+* AsyncMessenger: fix valgrind leak (`pr#7725 <http://github.com/ceph/ceph/pull/7725>`_, Haomai Wang)
+* Clarify usage on starting single osd/mds/mon. (`pr#7641 <http://github.com/ceph/ceph/pull/7641>`_, Patrick Donnelly)
+* Deprecate or free up a bunch of feature bits (`pr#8214 <http://github.com/ceph/ceph/pull/8214>`_, Samuel Just)
+* Event: fix clock skew problem (`pr#7949 <http://github.com/ceph/ceph/pull/7949>`_, Wei Jin)
+* FileStore: Added O_DSYNC write scheme (`pr#7752 <http://github.com/ceph/ceph/pull/7752>`_, Somnath Roy)
+* FileStore: fix initialization order for m_disable_wbthrottle (`pr#8067 <http://github.com/ceph/ceph/pull/8067>`_, Samuel Just)
+* Fix unicode encoding for non-ASCII characters in title and author names (`pr#7 <http://github.com/ceph/ceph/pull/7>`_, Abhishek Varshney)
```
Comment (theanalyst) [inline doc/release-notes.rst]: pr#7689 ?

Comment (dachary) [inline doc/release-notes.rst]: good catch

**Comment by dachary** [comment]
@theanalyst applied the suggested changes in a separate commit so we can know the difference between the script output **ceph-release-notes -r tags/v10.0.5..ceph/jewel $(git rev-parse --show-toplevel)** and the manual edits.

**Comment by theanalyst** [comment]
@dachary LGTM, maybe we should probably get this along with https://github.com/ceph/ceph/pull/8211 as well,

**Comment by liewegas** [comment]
manually  merged into wip-jewel

**Comment by liewegas** [comment]
https://github.com/ceph/ceph/pull/8211

**Pull request closed without merging**